input.go

```diff
@@ -289,3 +291,31 @@ func (ins Inputs) Less(i, j int) bool {
 func (ins Inputs) Swap(i, j int) {
 	ins[i], ins[j] = ins[j], ins[i]
 }
+
+// CreateVM ...
+func (ins Inputs) CreateVM(srcdir string) (*otto.Otto, error) {
+	self := map[string]map[string]interface{}{}
+	for _, i := range ins {
+		if i.Default != nil {
+			self[i.ID] = map[string]interface{}{
+				"path": i.Default.Entry.Location,
+			}
+		}
+		if i.Provided != nil {
```

Comment:
should be `early return`, or `early continue` in this case inside looping. If we prefer `Provided` than `Default`,

```
if Provided {
 // do it
 continue
}
if Default {
  // do another
  continue
}
// something if needed
```